File: steps/javascript/create-pet.step.js
// steps/javascript/create-pet.step.js
const { create } = require('./js-store');

exports.config = {
  type: 'api',
  name: 'JsCreatePet',
  path: '/js/pets',
  method: 'POST',
  emits: []
};

exports.handler = async (req) => {
  const b = req.body || {};
  const name = typeof b.name === 'string' && b.name.trim();
  const speciesOk = ['dog','cat','bird','other'].includes(b.species);
  const ageOk = Number.isFinite(b.ageMonths);
  if (!name || !speciesOk || !ageOk) {
    return { status: 400, body: { message: 'Invalid payload: {name, species, ageMonths}' } };
  }
  const pet = create({ name, species: b.species, ageMonths: Number(b.ageMonths) });
  return { status: 201, body: pet };
};

File: steps/python/create_pet.step.py
# steps/python/create_pet.step.py
config = {
    "type": "api",
    "name": "PyCreatePet",
    "path": "/py/pets",
    "method": "POST",
    "emits": []
}

async def handler(req, ctx=None):
    try:
        import sys
        import os
        sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
        from services import pet_store
    except ImportError:
        # Fallback for import issues
        return {"status": 500, "body": {"message": "Import error"}}
    
    b = (req.get("body") or {})
    name = b.get("name")
    species = b.get("species")
    age = b.get("ageMonths")
    if not isinstance(name, str) or not name.strip():
        return {"status": 400, "body": {"message": "Invalid name"}}
    if species not in ["dog","cat","bird","other"]:
        return {"status": 400, "body": {"message": "Invalid species"}}
    try:
        age_val = int(age)
    except Exception:
        return {"status": 400, "body": {"message": "Invalid ageMonths"}}
    pet = pet_store.create(name, species, age_val)
    return {"status": 201, "body": pet}
